app/src/main/java/com/example/homework3/fragments/DetailsFragment.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.homework3.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.navigation.ui.setupWithNavController
+import coil.load
+import com.example.homework3.retrofit.GithubService
+import com.example.homework3.databinding.FragmentDetailsBinding
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class DetailsFragment : Fragment() {
+
+    private var _binding: FragmentDetailsBinding? = null
+    private val binding
+        get() = requireNotNull(_binding) {
+            "View was destroyed"
+        }
+
+    private val args: DetailsFragmentArgs by navArgs()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentDetailsBinding.inflate(inflater, container, false)
+            .also { binding ->
+                _binding = binding
+            }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        loadUserDetails()
+
+        with(binding) {
+            toolbar.setupWithNavController(findNavController())
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun loadUserDetails() {
+        viewLifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
+            val userDetails =
+                GithubService.githubApi.getUserDetails(args.username)
+            with(binding) {
+                withContext(Dispatchers.Main) {
+                    avatar.load(userDetails.avatarUrl)
+                    login.text = userDetails.login
+                    followers.text = "Followers: ${userDetails.followers}"
+                    following.text = "Following: ${userDetails.following}"
+                }
+            }
+        }
```

Comment:
можно без Dispatchers.IO, retrofit самостоятельно переносит выполнение запроса на этот поток и потом результат у тебя будет в Dispatchers.Main
```suggestion
        viewLifecycleOwner.lifecycleScope.launch {
            val userDetails =
                GithubService.githubApi.getUserDetails(args.username)
            with(binding) {
                avatar.load(userDetails.avatarUrl)
                login.text = userDetails.login
                followers.text = "Followers: ${userDetails.followers}"
                following.text = "Following: ${userDetails.following}"
            }
        }
```